src/VisualStudio/Core/Def/Storage/AbstractCloudCachePersistentStorageService.cs

```diff
@@ -40,13 +39,17 @@ protected sealed override bool ShouldDeleteDatabase(Exception exception)
         protected sealed override async ValueTask<IChecksummedPersistentStorage?> TryOpenDatabaseAsync(
             SolutionKey solutionKey, string workingFolderPath, string databaseFilePath, CancellationToken cancellationToken)
         {
-            var cacheService = await this.CreateCacheServiceAsync(cancellationToken).ConfigureAwait(false);
+            var solutionFolder = IOUtilities.PerformIO(() => Path.GetDirectoryName(solutionKey.FilePath));
+
+            if (solutionFolder == null || solutionFolder == "")
+                return null;
```

Comment:
using `string.IsNullOrEmpty` does not seem to prove to the compiler that the string is non-null after this.

Reply:
(this file is shared, and it looks like in some configs this isn't nrt annotated).